src/internal/connector/onedrive/collection.go

```diff
@@ -129,37 +142,92 @@ func (oc *Collection) populateItems(ctx context.Context) {
 	defer colCloser()
 	defer close(folderProgress)
 
+	limitCh := make(chan struct{}, urlPrefetchChannelBufferSize)
+	defer close(limitCh)
+
+	type uerr struct {
+		itemID string
+		err    error
+	}
+
+	errCh := make(chan uerr, len(oc.driveItemIDs))
```

Comment:
I think the extra channel here adds some unnecessary complication.  Consider an alternative, that follows our other pattern of error aggregation with an anonymous func:
```go
var errs error
errUpdater := func(id string, err error) {
	errs = support.WrapAndAppend(id, err, errs)
}
```
This pattern removes the possibility of accidentally blocking on the error reporting channel (a common mistake when handling channels, which could appear in a future change, even if it's currently safe).  It would also reduce code by removing the need for the `ffail` counter, the `uerr` struct, and the loop at the end of the func.

Error aggregation on 189 could be reduced to:
```go
if err != nil {
  errUpdater(id, err)
}
```
Fast fail checks then on 158 turn into:
```go
if oc.service.ErrPolicy() && errs != nil {
```

Reply:
Maybe I am missing something but would this be concurrent safe? Since we are not likely to hit many errors, I can add add locks and update them in place as another alternative.

Reply:
Add mutexes as needed, sure.  In place vs calling an async func is just a matter of cleanliness/code reduction.

Reply:
I've added locks within `errUpdater`